homeworks/oleg.dzhygil_dzhygil/homework_5/homework_5.js

```diff
@@ -0,0 +1,86 @@
+// Task 1
+
+function countNumber() {
+    let number = 0;
+    return (value) => {
+        number += value;
+        return number;
+    };
+}
+
+function createMass(mass) {
+    const array = mass;
+    const summ = countNumber();
+    for (let i = 0; i < array.length; i++) {
+        array[i] = summ(array[i]);
+    }
+    return array;
+}
+
+console.log(createMass([10, -10, 10, -10, 10]));
+console.log(createMass([0, 0, 0, 0, 0]));
+console.log(createMass([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]));
+
+
+// Task 2
+
+const classes = ['header', 'menu', 'menu_item', 'tabs',
+    'tab_item', 'menu', 'link', 'tabs', 'tab_item', 'menu',
+    'menu_item', 'menu', 'menu_item'];
+
+
+function sortKey(arr) {
+    const obj = {};
+    arr.forEach((x) => {
+        if (x in obj) {
+            obj[x] += 1;
+        } else {
+            obj[x] = 1;
+        }
+    });
+    return obj;
+}
+console.log(sortKey(classes));
+
+// Task 3
+
+const obj = { b: 'c', d: { e: 'f' } };
+
+function innerCopy() {
+    const newObject = {};
+    Object.keys(obj).forEach((key) => {
+        newObject[key] = obj[key];
+    });
+    return newObject;
```

Comment:
this will not perform deep object copy, read about variables as references to objects to know why